buildbot.mariadb.org/scripts/deb-upgrade.sh

```diff
@@ -0,0 +1,449 @@
+if [[ $test_mode == "all" ]]; then
+  if dpkg -l | grep -iE 'mysql|maria' | grep "$(cat /tmp/version.old)"; then
+    bb_log_err "Old packages have been found after upgrade"
+    exit 1
+  fi
+fi
+
+#=====================================================================================
+# Check that the server is functioning and previously created structures are available
+#=====================================================================================
+
+# Print "have_xx" capabilitites for the new server
+mysql -uroot -prootpass -e "select 'Stat' t, variable_name name, variable_value val from information_schema.global_status where variable_name like '%have%' union select 'Vars' t, variable_name name, variable_value val from information_schema.global_variables where variable_name like '%have%' order by t, name"
+
+# All the commands below should succeed
+set -e
+mysql -uroot -prootpass -e "select @@version, @@version_comment"
+mysql -uroot -prootpass -e "SHOW TABLES IN db"
+mysql -uroot -prootpass -e "SELECT * FROM db.t_innodb; INSERT INTO db.t_innodb VALUES (3,'foo'),(4,'bar')"
+mysql -uroot -prootpass -e "SELECT * FROM db.t_myisam; INSERT INTO db.t_myisam VALUES (3,'foo'),(4,'bar')"
+mysql -uroot -prootpass -e "SELECT * FROM db.t_aria; INSERT INTO db.t_aria VALUES (3,'foo'),(4,'bar')"
+echo "If the next INSERT fails with a duplicate key error,"
+echo "it is likely because the server was not upgraded or restarted after upgrade"
+mysql -uroot -prootpass -e "SELECT * FROM db.t_memory; INSERT INTO db.t_memory VALUES (1,'foo'),(2,'bar')"
+mysql -uroot -prootpass -e "SELECT COUNT(*) FROM db.v_merge"
+mysql -uroot -prootpass -e "SELECT COUNT(*) FROM db.v_temptable"
+mysql -uroot -prootpass -e "CALL db.p()"
+mysql -uroot -prootpass -e "SELECT db.f()"
+
+if [[ $test_mode == "columnstore" ]]; then
+  if ! mysql -uroot -prootpass -e "SELECT * FROM db.t_columnstore; INSERT INTO db.t_columnstore VALUES (3,'foo'),(4,'bar')"; then
+    get_columnstore_logs
+    exit 1
+  fi
+fi
+
+#===================================================================================
+# Store information about server version and available plugins/engines after upgrade
+#===================================================================================
+mysql -uroot -prootpass --skip-column-names -e "select @@version" | awk -F'-' '{ print $1 }' >/tmp/version.new
+mysql -uroot -prootpass --skip-column-names -e "select engine, support, transactions, savepoints from information_schema.engines" | sort >/tmp/engines.new
+case "$major_version" in
+  5.5)
+    mysql -uroot -prootpass --skip-column-names -e "show plugins" | sort >/tmp/plugins.new
+    ;;
+  10.[0-9])
+    mysql -uroot -prootpass --skip-column-names -e "select plugin_name, plugin_status, plugin_type, plugin_library, plugin_license from information_schema.all_plugins" | sort >/tmp/plugins.new
+    ;;
+esac
+
+# //TEMP what needs to be done here?
```

Comment:
here there are some dependencies lists that are checked in the old setup (https://github.com/MariaDB/mariadb.org-tools/tree/master/buildbot/baselines). Not sure what we should do in the new version.

Reply:
Yep, me neither... 
I have just created https://jira.mariadb.org/browse/MDBF-401